### ISSUE TITLE
Measure simulation is breaking measure aggregation

### ISSUE THREAD

**Issue by Harold7665** [atoti/atoti#122]
## Steps to reproduce
```python
import atoti as tt
import pandas as pd

data = [
    ['2019-06-30', 'Germany', 'CCCCCCCCCCCCQB999NBRG', 0.01],
    ['2019-06-30', 'Germany', 'CCCCCCCCCCCCQB99M9BRG', 0.063],
    ['2019-06-30', 'Germany', 'CCCCCCCCCCCCQB99MBJRG', 0.067]
]
df = pd.DataFrame(data, columns=['Reporting Date', 'Legal Entity', 'Contract ID', 'PD12'])

session = tt.create_session()

contracts_store = session.read_pandas(df, keys=['Reporting Date', 'Legal Entity', 'Contract ID'], store_name='Store')
cube = session.create_cube(contracts_store, 'IFRS9', mode='no_measures')
h = cube.hierarchies
l = cube.levels
m = cube.measures

h['Legal Entity'].slicing = True

m['PD (12)'] = tt.agg.mean(contracts_store['PD12'])
m['PD (12) mean'] = tt.agg.mean(m['PD (12)'], scope=tt.scope.origin(l['Contract ID']))

session.query_mdx('SELECT NON EMPTY {[Measures].[PD (12)], [Measures].[PD (12) mean]} ON COLUMNS, NON EMPTY [Hierarchies].[Contract ID].[ALL].[AllMember] ON ROWS FROM [IFRS9] WHERE [Hierarchies].[Reporting Date].[ALL].[AllMember].[2019-06-30]')
```
```python
pd_12_simulation = cube.setup_simulation(
    'PD (12) scaling simulation',
    levels=[l['Reporting Date'], l['Legal Entity'], l['Contract ID']],
    multiply=[m['PD (12)']]
)
```
```python
session.query_mdx('SELECT NON EMPTY {[Measures].[PD (12)], [Measures].[PD (12) mean]} ON COLUMNS, NON EMPTY [Hierarchies].[Contract ID].[ALL].[AllMember] ON ROWS FROM [IFRS9] WHERE [Hierarchies].[Reporting Date].[ALL].[AllMember].[2019-06-30]')
```

## Actual Result
SUM aggregation on PD (12).

## Expected Result
MEAN aggregation on PD (12).

## Environment
- atoti: 0.4.3.20200724145216
- Python: 3.8.5
- Operating System: win32

**Comment by Skyscimitar** [comment]
This was due to an error when creating simulation measures, we ignored the previously defined aggregation function.
This behavior will be fixed in the next release.

**Comment by Harold7665** [comment]
Hello,
After your fix I got the right aggregation, but it came back on this version:
- atoti: 0.4.3.20200803144311
- Python: 3.8.5
- Operating System: win32

**Comment by Skyscimitar** [comment]
I wasn't able to reproduce the error with the provided example using the latest version of atoti. Please update the example, or provide an notebook which reproduces the error. Or open a new ticket if the error is different.

**Comment by Skyscimitar** [comment]
I investigated the issue using the full notebook. The disparity in results before and after the simulation are actually not due to the simulation itself. When creating a simulation, this changes the schema of the cube/datastore, which triggers a reload of the data.
The sampling mode used in your notebook is `first_lines: 5`, which loads the first five lines of the first file found in the directory. However, a DirecotryTopic uses java `DirecotryStream`s to list files, which do not garanty any order in the returned files. The differences in the values you still see are in fact due to the lines loaded before and after the simulation being different.

**Comment by Skyscimitar** [comment]
Even further investigation showed that there was also still a problem with the simulation measures. The problem appeared when defining several simulations, and using multilevel-hierarchies at the same time. We failed to correctly transmit the aggregation function of the new simulation measure, which are re-created each time new measures are published, which meant that immediately after creating the simulation the measure worked as expected, but after making changes to the cube it would break again.

This should be working properly again once the pr activeviam/atoti/#1547 is merged.
